src/commands/logistics/netid.ts

```diff
@@ -0,0 +1,115 @@
+            interaction.reply('The given netID is invalid. Usage: `/netid <netid>`');
+            return;
+        }
+
+        // Check if netid is already verified
+        let user = User.findOne({ netid: netid }).catch(err => {
+            console.log(err);
+            interaction.reply(ERROR_MSG)
+        })
+        if (user) {
+            if (user.verified) {
+                interaction.reply('You are already verified!');
+                return;
+            }
+        } else {
+            user = new User({ discordId: interaction.user.id, netid: null, verified: false });
+            await user.save();
+        }
+
+        // Timeout the command for each NetID and each user to prevent spam
+        const activeTokens = await Verification.find({
+            $or: [
+                { netid: netid, tokenExpiration: { $gt: Date.now() } },
+                { discordId: interaction.user.id, tokenExpiration: { $gt: Date.now() } }]
+        }).exec();
+
+        if (activeTokens.length > 0) {
+            interaction.reply('You have already requested a verification token. Please wait 30 minutes before requesting another.');
+            return;
+        }
+
+        // Check if user is already verified
+        User.findOne({ discordId: interaction.user.id }, (err, user) => {
+            if (err) {
+                interaction.reply(ERROR_MSG)
+            } else if (user) {
+                if (user.verified) {
+                    interaction.reply('You are already verified!');
+                }
+            }
+        });
+
+        const verification = await createVerification(netid, interaction.user.id);
+
+        await verification.save().catch(err => {
+            console.log(err);
+            interaction.reply(ERROR_MSG)
+        });
+
+        const sgMail = require('@sendgrid/mail')
+        sgMail.setApiKey(process.env.SENDGRID_API_KEY)
+
+        await sgMail.send({
+            to: netid + '@illinois.edu',
+            from: 'noreply@opensourceatillinois.com',
+            subject: 'Email Verification: Open-Source @ Illinois',
+            html: getEmailString(verification.token),
+        }).catch((error) => {
+            console.error(error)
+            interaction.reply(ERROR_MSG);
+        });
+
+        interaction.reply('You should receive a verification code shortly. Use /verify to complete the process.');
+    },
+};
+
+const createVerification = async (netid: string, discordId: string) => {
+    const token = await getRandomToken();
+    // Set expiration to 30 minutes from now
+    const verification = new Verification({
+        netid: netid,
+        discordId: discordId,
+        token: token,
+        tokenExpiration: Date.now() + VERIFY_TOKEN_EXPIRATION,
+    });
+    return verification;
+}
+
+const getRandomToken = async () => {
+    const response = await got.get(`https://www.random.org/strings/?num=1&len=${VERIFY_TOKEN_LENGTH}&digits=on&upperalpha=on&unique=on&format=plain&rnd=new`)
```

Comment:
Is calling to random.org necessary?

Reply:
Not strictly. I figured this was the best way to avoid pseudo-random generation that can sometimes be vunerable